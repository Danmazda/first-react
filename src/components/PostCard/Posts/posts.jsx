import { PostCard } from "..";
import './styles.css';


export const Posts = ({posts})=>(
    <div className='posts'>
    {posts.map(post =>(
      //Todos os elementos do map precisam de key(id)
      // Componente precisa ser aberto e fechado como uma tag html
      // atributos da tag ser√£o passados como props
      <PostCard 
      key={post.id}
      title={post.title}
      cover ={post.cover}
      body = {post.body}
      />
    ))}
  </div>
); 