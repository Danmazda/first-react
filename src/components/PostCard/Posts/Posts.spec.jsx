import { render, screen } from '@testing-library/react';
import { Posts } from './posts';
// key={post.id}
// title={post.title}
// cover ={post.cover}
// body = {post.body}
const props = [
  {
    id: 1,
    title: 'title 1',
    cover: 'img/img.png',
    body: 'body 1',
  },
];

describe('<Posts />', () => {
  it('should render Posts', () => {
    render(<Posts posts={props} />);
    expect(screen.getByRole('heading', 'title 1')).toBeInTheDocument();
    expect(screen.getByAltText('title 1')).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });
  it('should be like the snapshot', () => {
    const { container } = render(<Posts posts={props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
