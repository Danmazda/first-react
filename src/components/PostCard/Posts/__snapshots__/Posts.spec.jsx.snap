// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should be like the snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img/img.png"
    />
    <h1>
      title 1
    </h1>
    <p>
      body 1
    </p>
  </div>
</div>
`;
